demo/c/rhino_demo_file.c

```diff
@@ -165,38 +223,54 @@ int main(int argc, char *argv[]) {
         exit(1);
     }
 
-    pv_rhino_t *rhino = NULL;
-    pv_status_t status = pv_rhino_init_func(model_path, context_path, 0.5f, &rhino);
-    if (status != PV_STATUS_SUCCESS) {
-        fprintf(stderr, "'pv_rhino_init' failed with '%s'\n", pv_status_to_string_func(status));
+    drwav f;
+
+    if (!drwav_init_file(&f, wav_path, NULL)) {
+        fprintf(stderr, "failed to open wav file at '%s'.", wav_path);
         exit(1);
     }
 
-    FILE *wav = fopen(wav_path, "rb");
-    if (!wav) {
-        fprintf(stderr, "failed to open wav file\n");
+    if (f.sampleRate != (uint32_t) pv_sample_rate_func()) {
+        fprintf(stderr, "audio sample rate should be %d\n.", pv_sample_rate_func());
         exit(1);
     }
 
-    if (fseek(wav, 44, SEEK_SET) != 0) {
-        printf("failed to skip the wav header\n");
+    if (f.bitsPerSample != 16) {
+        fprintf(stderr, "audio format should be 16-bit\n.");
         exit(1);
     }
 
-    const int32_t frame_length = pv_rhino_frame_length_func();
+    if (f.channels != 1) {
+        fprintf(stderr, "audio should be single-channel.\n");
+        exit(1);
+    }
 
-    int16_t *pcm = malloc(sizeof(int16_t) * frame_length);
+    int16_t *pcm = calloc(pv_rhino_frame_length_func(), sizeof(int16_t));
     if (!pcm) {
-        fprintf(stderr, "failed to allocate memory for audio buffer\n");
+        fprintf(stderr, "failed to allocate memory for audio frame.\n");
         exit(1);
     }
 
-    double total_cpu_time_usec = 0;
-    double total_processed_time_usec = 0;
+    pv_rhino_t *rhino = NULL;
+    pv_status_t status = pv_rhino_init_func(
+            access_key,
+            model_path,
+            context_path,
+            sensitivity,
+            require_endpoint,
+            &rhino);
+    if (status != PV_STATUS_SUCCESS) {
+        fprintf(stderr, "'pv_rhino_init' failed with '%s'\n", pv_status_to_string_func(status));
+        exit(1);
```

Comment:
dont we need to do some clean up when exiting the program?

Reply:
good question - none of our other C demos do cleanup on early exit, so I assumed it was decided that was unnecessary. Are we worried about lost memory? I did some googling and it seemed the consensus was that the OS reclaims memory from any process that exits.

Reply:
correct, and as I mentioned I'm not sure what's the best practice here, just raised a concern :)